dascore/io/dasvader/utils.py

```
@@ -77,15 +76,18 @@ def _raise_legacy_ref_error(h5, field_name: str) -> None:
         return value
-    if h5r.get_name(value, h5.id) is None:
-        _raise_legacy_ref_error(h5, field_name)
     try:
         return h5[value]
     except KeyError:
-        return h5py.Dataset(dereference(value, h5.id))
+        # The high-level lookup fails for some references HDF5 can still
+        # resolve directly, so try that before giving up.
+        try:
+            return h5py.Dataset(dereference(value, h5.id))
+        except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)